### PR TITLE
Fix crash on loading unimported texture.

### LINKED ISSUE #323: Able to add not imported texture to material component
Steps to reproduce:
1. Create a texture without importing it
2. Create a entity with a material component
3. Add the un-imported texture to the material
4. Crash!

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
A new texture now defaults to the default texture.
Fixes #323 .

### CODE REVIEW

File: src/Engine/Manager/ResourceManager.cpp [hunk 1]
```diff
@@ -74,6 +74,21 @@ TextureAsset* ResourceManager::CreateTextureAsset(const std::string& name) {
...
+    if (textureAssets.find(name) == textureAssets.end()) {
+        TextureAsset* textureAsset = new TextureAsset();
+        textureAsset->Load(name, texture);
+        textureAssets[name].textureAsset = textureAsset;
+        textureAssetsInverse[textureAsset] = name;
+        textureAssets[name].count = 1;
+    }
+    else {
+        textureAssets[name].count++;
+    }
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ResourceManager.cpp]: MINOR The function 'GetTextureAssetInstanceCount' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ResourceView.cpp#L23](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/56fcda7ec56a5594f0cbb71a30bd934dc93c5858/src/Editor/GUI/ResourceView.cpp#L23): Member variable 'ResourceView::scriptPressed' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [ResourceView.cpp#L23](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/56fcda7ec56a5594f0cbb71a30bd934dc93c5858/src/Editor/GUI/ResourceView.cpp#L23): Member variable 'ResourceView::texturePressed' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [ResourceView.cpp#L23](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/56fcda7ec56a5594f0cbb71a30bd934dc93c5858/src/Editor/GUI/ResourceView.cpp#L23): Member variable 'ResourceView::modelPressed' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [ResourceView.cpp#L23](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/56fcda7ec56a5594f0cbb71a30bd934dc93c5858/src/Editor/GUI/ResourceView.cpp#L23): Member variable 'ResourceView::soundPressed' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Texture/TextureAsset.hpp [hunk 2]
```diff
@@ -24,6 +24,14 @@ class TextureAsset {
          * @param name The name of the texture asset.
          */
         void Load(const std::string& name);
+
+        /// Load texture asset from memory.
+        /**
+        * @param name The name of the texture asset.
+        * @param texture The texture we will load our texture information from.
+        * @param srgb Whether to convert from SRGB space to linear space when loading.
+        */
```
Comment (Sajdkick) [inline src/Engine/Texture/TextureAsset.hpp]: Spaces for '*' to align

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**